pkg/execution/state/redis_state/queue_processor.go

```diff
@@ -428,7 +442,7 @@ func (q *queue) processPartition(ctx context.Context, p *QueuePartition, f osque
 			case ErrPartitionConcurrencyLimit, ErrConcurrencyLimit:
 				q.scope.Counter(counterConcurrencyLimit).Inc(1)
 				q.sem.Release(1)
-				return nil
+				return err
```

Comment:
```suggestion
				// Since the queue is at capacity, return the error so that we
				// don't keep hammering with "does the queue have room?" logic.
				return err
```